app/assets/javascripts/homepage/conversation.coffee

```diff
@@ -235,3 +259,31 @@ conversation.controller("ConversationController", ["$scope", "$stateParams", "AP
 
     parseConversation(ConversationData.data, true)
 ])
+
+conversation.service("ConversationService", () ->
+    conversations = {}
+    getConversation = (conversationId) ->
+        if conversationId of conversations
+            return conversations[conversationId]
+        else
+            c = {addPostText: "", summaryText: "", resolutionText: ""}
+            conversations[conversationId] = c
+            return c
+    # getter and setter for posts in progress
+    this.updateAddPostText = (conversationId, text) ->
+        getConversation(conversationId).addPostText = text
+    this.updateSummaryText = (conversationId, text) ->
+        getConversation(conversationId).summaryText = text
+    this.updateResolutionText = (conversationId, text) ->
+        getConversation(conversationId).resolutionText = text
+    this.getAddPostText = (conversationId) ->
+        return getConversation(conversationId).addPostText
+    this.getSummaryText = (conversationId) ->
+        return getConversation(conversationId).summaryText
+    this.getResolutionText = (conversationId) ->
+        return getConversation(conversationId).resolutionText
+    this.hasUnsavedProgress = (conversationId) ->
+        c = getConversation(conversationId)
+        return (c.addPostText || c.summaryText || c.resolutionText)
```

Comment:
Heads up: this returns a string, not a boolean. It probably works still, but I'd rather it be `c.addPostText != "" or .....`